libraries/standard/http/include/http_client.h

```diff
@@ -130,12 +130,12 @@
 #define HTTP_RANGE_REQUEST_END_OF_FILE              -1
 
 /**
- * @brief The HTTPNetworkContext is an incomplete type. The application must
- * define HTTPNetworkContext to the type of their network context. This context
+ * @brief The NetworkContext is an incomplete type. The application must
+ * define NetworkContext to the type of their network context. This context
  * is passed into the network interface functions.
  */
-struct HTTPNetworkContext;
-typedef struct HTTPNetworkContext HTTPNetworkContext_t;
+struct NetworkContext;
+typedef struct NetworkContext * NetworkContext_t;
```

Comment:
This is defined here for now but will be transferred once we make the common transport library. Also, I still think it's a little weird that this is the only struct ptr type in the entire library, but that's just me.

Reply:
No, I think it's pretty weird too

Reply:
It follows what FreeRTOS kernel does with opaque types. This is to void pointers.

Reply:
Looking at the entire PR now, I think, HTTP library itself should not reinterpret NetworkContext_t , it should be defined in network_interface.h (eventually). Make sure the implementation is same as MQTT, since we are expected to share the transport implementation.